cpp/include/cudf/copying.hpp

```diff
@@ -49,13 +50,89 @@ namespace experimental {
  * @param[in] check_bounds Optionally perform bounds checking on the values
  * of `gather_map` and throw an error if any of its values are out of bounds.
  * @params[in] mr The resource to use for all allocations
- * @return cudf::table Result of the gather
+ * @return std::unique_ptr<table> Result of the gather
  */
 std::unique_ptr<table> gather(table_view const& source_table, column_view const& gather_map,
 			      bool check_bounds = false,
 			      rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
-			       
+/**
+ * @brief Creates a new table that results from scattering values from the rows
+ * of a source table into the rows of a target table according to a scatter map.
+ *
+ * Scatters values from the source table into the target table out-of-place,
+ * returning a "destination table". The scatter is performed according to a scatter map
+ * such that row "i" of the source table will be equal to row "scatter_map[i]"
+ * of the destination table. All other rows of the destination table will be equal to
+ * corresponding rows of the target table.
+ *
+ * The datatypes between coresponding columns in the source and destination
+ * must be the same.
+ *
```

Comment:
There should be a note about how negative values are handled (similar to the gather docs).

Reply:
Also add a note about if a value is repeated in `scatter_map` behavior is undefined.